sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.h

```diff
@@ -0,0 +1,476 @@
+ */
+#define _NIB_IF_MASK        (GNRC_IPV6_NIB_NC_INFO_IFACE_MASK)
+
+/**
+ * @brief   Shorter name for convenience ;-)
+ */
+#define _NIB_IF_POS         (GNRC_IPV6_NIB_NC_INFO_IFACE_POS)
+
+/**
+ * @brief   On-link NIB entry
+ */
+typedef struct _nib {
+    struct _nib *next;                  /**< next removable entry */
+#if GNRC_IPV6_NIB_CONF_QUEUE_PKT || defined(DOXYGEN)
+    /**
+     * @brief   queue for packets currently in address resolution
+     *
+     * @note    Only available if @ref GNRC_IPV6_NIB_CONF_QUEUE_PKT != 0
+     */
+    gnrc_pktqueue_t *pktqueue;
+#endif
+    /**
+     * @brief Neighbors IPv6 address
+     */
+    ipv6_addr_t ipv6;
+#if GNRC_IPV6_NIB_CONF_6LN || defined(DOXYGEN)
+    /**
+     * @brief   The neighbors EUI-64 (used for DAD)
+     *
+     * @note    Only available if @ref GNRC_IPV6_NIB_CONF_6LN != 0
+     */
+    eui64_t eui64;
+#endif
+#if GNRC_IPV6_NIB_CONF_ARSM || defined(DOXYGEN)
+    /**
+     * @brief   Link-layer address of _nib_t::next_hop
+     *
+     * @note    Only available if @ref GNRC_IPV6_NIB_CONF_ARSM != 0
+     */
+    uint8_t l2addr[GNRC_IPV6_NIB_L2ADDR_MAX_LEN];
+#endif
+
+    /**
+     * @brief   Information flags (see @ref net_gnrc_ipv6_nib_nc_info "NC info values")
+     */
+    uint16_t info;
+    uint8_t mode;                       /**< NIB mode */
+#if GNRC_IPV6_NIB_CONF_ARSM || defined(DOXYGEN)
+    /**
+     * @brief   Neighbor solicitations sent for probing
+     *
+     * @note    Only available if @ref GNRC_IPV6_NIB_CONF_ARSM != 0
+     */
+    uint8_t ns_sent;
+
+    /**
+     * @brief   length of _nib_t::l2addr
+     *
+     * @note    Only available if @ref GNRC_IPV6_NIB_CONF_ARSM != 0
+     */
+    uint8_t l2addr_len;
+#endif
+} _nib_t;
+
+/**
+ * @brief   Default route NIB entry
+ */
+typedef struct {
+    _nib_t *next_hop;   /**< next hop to destination */
+    uint16_t ltime;     /**< lifetime in seconds */
+} _nib_dr_t;
+
+/**
+ * @brief   Off-link NIB entry
+ */
+typedef struct {
+    _nib_t *next_hop;           /**< next hop to destination */
+    ipv6_addr_t pfx;            /**< prefix to the destination */
+    unsigned pfx_len;           /**< prefix-length of _nib_t::dst */
+} _nib_dst_t;
+
+/**
+ * @brief   Interface specific information for Neighbor Discovery
+ */
+typedef struct {
+#if GNRC_IPV6_NIB_CONF_ARSM
+    uint32_t reach_time_base;           /**< base for random reachable time calculation */
+    uint32_t reach_time;                /**< reachable time (in ms) */
+    uint32_t retrans_time;              /**< retransmission time (in ms) */
+#endif
+#if GNRC_IPV6_NIB_CONF_ROUTER || defined(DOXYGEN)
+    /**
+     * @brief   timestamp in milliseconds of last unsolicited router advertisement
+     *
+     * @note    Only available if @ref GNRC_IPV6_NIB_CONF_ROUTER
+     */
+    uint32_t last_ra;
+#endif
+    kernel_pid_t pid;                   /**< identifier of the interface */
+#if GNRC_IPV6_NIB_CONF_ROUTER || defined(DOXYGEN)
```

Comment:
Yes, and I think this has more precedence than readability in a *constraint* environment. Especially when it comes to *internal* types.